### DISCUSSION TITLE
Support for websocket middleware

### DISCUSSION THREAD

**Discussion by DrPyser** [encode/starlette#1192] [Q&A]
Is there any plan to support middlewares for websocket connections? 

The same use cases for middlewares on http connections can also apply on websockets, such as dealing with managing lifecycle of request/connection-bound resources.

**Comment by lovelydinosaur** [comment]
ASGI Middleware already supports websockets, yup, and some of our middleware classes apply to both websockets and http.

What we don't yet have is an equivelent to `BaseHTTPMiddleware`. We could probably have a `BaseConnectionMiddleware` which applies to any connection, either http or websockets. (`call_next

**Comment by abompard** [comment]
I would be interested in this `BaseConnectionMiddleware`, if you end up adding it. Thanks.

**Comment by ajoino** [comment]
> ASGI Middleware already supports websockets, yup, and **_some of our middleware classes apply to both websockets and http_**.


What is it that makes your middleware work on both WS and HTTP and even though it doesn't work for end users?

**Comment by demian-sl** [comment]
upvote
